pymc3/model.py

```diff
@@ -972,7 +976,37 @@ def cont_vars(self):
         """All the continuous variables in the model"""
         return list(typefilter(self.vars, continuous_types))
 
-    def Var(self, name, dist, data=None, total_size=None):
+    def shape_from_dims(self, dims):
+        shape = []
+        if len(set(dims)) != len(dims):
+            raise ValueError('Can not contain the same dimension name twice.')
+        for dim in dims:
+            if dim not in self.coords:
+                raise ValueError('Unknown dimension name %s. All dimension '
+                                 'names must be specified in the `coords` '
+                                 'argument of the model or through a pm.Data '
+                                 'variable.' % dim)
+            shape.extend(self.coords[dim].shape)
+        return tuple(shape)
+
+    def add_coords(self, coords):
+        if coords is None:
+            return
+
+        for name in coords:
+            if not name.isidentifier():
+                raise ValueError('Invalid dimension name. Must be a python '
+                                 'identifier: %s' % name)
```

Comment:
- Possible to Blackify this?
- Maybe use a slightly less cryptic message -- I'm not sure most people know what a python identifier is

Reply:
I'm changing it to
```python
raise ValueError(
    "Dimension names should be Python identifiers (alphanumeric with underscores, not starting with a number)."
    " The passed value was: %s" % name
)
```

Reply:
👍